pfrl/collections/prioritized.py

```diff
@@ -222,7 +259,7 @@ def _find(index_left, index_right, node, pos):
             return _find(index_center, index_right, node_right, pos - left_value)
 
 
-class SumTreeQueue(TreeQueue):
+class SumTreeQueue(TreeQueue[float]):
```

Comment:
I'm slightly confused here, so maybe I misunderstand. Why do you have `TreeQueue[float]` for `SumTreeQueue` and `MinTreeQueue`? I'm looking at the documentation (https://docs.python.org/3/library/typing.html) under "User-defined generic types" and it says: "Generic[T] as a base class defines that the class LoggedVar takes a single type parameter T". Is this because you're saying that a `SumTreeQueue` is a `TreeQueue` where the Generic type V (from `class TreeQueue(Generic[V])`) is specifically a float in this case?

Reply:
Yes, `SumTreeQueue` extends `TreeQueue` where `V` is `float`. SumTreeQueue's methods assume this, e.g. `max` returns 0.0 when empty.